CMakeLists.txt

```diff
@@ -214,25 +214,6 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_CONFIGURATION_TYPES)
   check_and_add_compiler_option(-Wused-but-marked-unused) # Clang/llVM only
 endif()
 
-## append ASAN build flags if compiler version has support
-#if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
-#   if(CMAKE_C_COMPILER_VERSION VERSION_GREATER 4.8)
-#      set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fsanitize=address
-#         -fno-omit-frame-pointer" CACHE STRING "" FORCE)
-#      message("Building with ASAN support (GNU compiler)")
-#   else()
-#      message("Building without ASAN support (GNU compiler)")
-#   endif()
-#elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
-#   if(CMAKE_C_COMPILER_VERSION VERSION_GREATER 3.1)
-#      set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fsanitize=address
-#         -fno-omit-frame-pointer" CACHE STRING "" FORCE)
-#      message("Building with ASAN support (Clang compiler)")
-#   elseif(CMAKE_C_COMPILER_VERSION VERSION_GREATER 3.1)
-#      message("Building without ASAN support (Clang compiler)")
-#   endif()
-#endif()
-
```

Comment:
This is / was meant only for debugging. So to easily be able to enable ASAN support for troubleshooting memory issues.
Is this somewhere else able to be just, switched on?

Reply:
I'll try to add these features as an independent make target.
I saw something similar before so it should be possible.

> I think one TODO is enough.

Me too, I hope I will be able to delete my todo by the end of this PR.
Wish me luck 😄